methods/enumerables.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index], index
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    answer = []
+    my_each do |item|
+      answer.push(item) if yield item
+    end
+    answer
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
```

Comment:
## my_all?:

- returns false if the block returns false or nil:

```ruby
    
    range = Range.new(5, 50)
    false_block = proc { |num| num > 88 }
    range.my_all?(&false_block) === range.all?(&false_block) => false

```

- when no block or argument is given returns false when one of the collection members are false or nil:

```ruby
    
    false_array = [nil, false, 1, [], 'hi']
    false_array.my_all? === false_array.all? => false

```

- when a class is passed as an argument returns true if all of the collection is a member of such class::Integer:

```ruby
    
    false_array = [1,2,3,4,32,6]
    array.my_all?(Integer) === array.all?(Integer) => false

```

- when a class is passed as an argument returns false if any of the collection is not a member of such class.

- when a Regex is passed as an argument returns true if all of the collection matches the Regex:

```ruby
    
    words = %w[dog door rod blade]
    words.my_all?(/d/) === words.all?(/d/) => false

```

- when a Regex is passed as an argument returns false if any of the collection does not match the Regex.

- when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern:

```ruby
    
    array = [3,4 ,43,2,1]
    array.my_all?(3) === array.all?(3) => false

```

- when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern.